### PR TITLE
Update CI to enable net7.0 preview builds

### PR DESCRIPTION
**Pull request by JialinXin**
### Summary of the changes (Less than 80 chars)
 - Detail 1
 - Detail 2

Fixes #bugnumber (in this specific format)

### CODE REVIEW

File: test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs [hunk 1]
```diff
@@ -1347,7 +1350,8 @@ public async Task TestEndpointManagerWithMultiHubsWithServerEndpointUpdate()
             await Task.WhenAll(containers.Select(x => x.MockReceivedServersPing("aaa;bbb")));
             containers1 = container1.GetTestOnlineContainers();
             await Task.WhenAll(containers1.Select(x => x.MockReceivedServersPing("aaa;bbb")));
-            await Task.Delay(6000);
+            await hubEndpoints.Single(x => x.Name == "1" && x.ServerEndpoint.AbsoluteUri == testSe).ScaleTask;
+            await Task.Delay(100);
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs:1354]: Is there still need to delay?

File: test/Microsoft.Azure.SignalR.Tests/Microsoft.Azure.SignalR.Tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Tests/Microsoft.Azure.SignalR.Tests.csproj:4]: hm.. I think we might need to separate net7 support to another  PR or maybe in the same PR supporting client invocation, we need to make sure net7 features like client invocation are supported before we claim that the package now supports net7

File: test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs (hunk 1 above)
Comment (JialinXin) [reply test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs:1354]: Yes. Little delay required as there's a few operations after the `ScaleTask` completes which is to sync the endpoints changes in the other side. So minor delay is preferred to ensure the validation success.

File: test/Microsoft.Azure.SignalR.Tests/Microsoft.Azure.SignalR.Tests.csproj (hunk 2 above)
Comment (JialinXin) [reply test/Microsoft.Azure.SignalR.Tests/Microsoft.Azure.SignalR.Tests.csproj:4]: OK.

File: test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs [hunk 3]
```diff
@@ -1050,8 +1050,8 @@ public async Task TestEndpointManagerWithAddEndpointsWithServersTag()
             var serversTag = "Server1;Server2;Server3";
             await Task.WhenAll(containers.Select(c => c.MockReceivedServersPing(serversTag)));
 
-            // wait one interval+ for Ready state and check negotiation is added.
-            await Task.Delay(6000);
+            await hubEndpoints[1].ScaleTask;
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs:1053]: await hubEndpoints[1].ScaleTask.OrTimeout(10000) ?

Comment (JialinXin) [reply test/Microsoft.Azure.SignalR.Tests/MultiEndpointServiceConnectionContainerTests.cs:1053]: `ScaleTask` is completed, but other operations need a few moment delay before negotiation endpoints get updated. I think `OrTimeout()` means the longest time that should be await when target task doesn't complete as expected.

In normal case it takes about 5.2s of an interval. Do you mean add `OrTimeout()` to make it fail fast in error case?

**Review by vicancy — APPROVED** [review]

**Pull request merged**